### PR TITLE
updated Ames packet format

### PR DESCRIPTION
**Pull request by drbeefsupreme**
This PR updates the Ames overview page to reflect the packet structure
introduced with https://github.com/urbit/urbit/pull/4051 and to address #995.

Reading the code, the format of the payload did not seem to be the same as that
written in the PR description, unless I'm misunderstanding something:

https://github.com/urbit/urbit/blob/master/pkg/arvo/sys/vane/ames.hoon#L421-L422

So I've written it as

```
 - 4 bits: sender life (mod 16)
 - 4 bits: receiver life (mod 16)
 - variable: sender address
 - variable: receiver address
 - 48 bits: (optional) 48-bit `origin`
 - 128 bits: `SIV`
 - 16 bits: ciphertext size
 - variable: ciphertext
 ```

rather than

```
48  bits:  (optional) 48-bit origin
4   bits:  sender life (mod 16)
4   bits:  receiver life (mod 16)
<variable> sender address
<variable> receiver address
128 bits:  SIV
16  bits:  ciphertext size
<variable> ciphertext
```

as indicated in the PR. Please advise if I am mistaken.

This also required modifying the example scenario towards the end of the
document, as well as the accompanying diagram. I added in a bit more detail
about how multi-packet messages are handled. Please read carefully to make sure
I did not misunderstand how relaying works, especially how the SIV is
incorporated since I found it a little confusing to say that the SIV is part of
the packet structure but its never actually sent over the wire.

Lastly, this links to new pages in #1091, so it should not be merged until after
#1091 is.

### CODE REVIEW

**Comment by drbeefsupreme** [comment]
My original comment truncated the part of the packet format that looked different (the origin), making it look like I was saying they were different and then presenting two identical formats. I thought I had gone insane for a moment. Fixed.

**Comment by drbeefsupreme** [comment]
@belisarius222 have you had time to look at this?

File: arvo/ames/ames.md [hunk 1]
```diff
@@ -319,34 +334,45 @@ with `%boon`s and create congestion there. So naxplanations have their own bone,
...
-future, galaxies will only need to know the star sponsoring a planet, and stars
-will be responsible for the final step of the relay.
+future, galaxies will only need to know the IP and port of the star sponsoring a
+planet, and stars will be responsible knowing the IP and port of their sponsored planets.
+
+Galaxies are also utilized for packet relaying in the case where two ships
+cannot communicate directly with one another, as is often the case when one or
+both are behind a
+[NAT](https://en.wikipedia.org/wiki/Network_address_translation) or certain
+firewalls. Again, stars will someday assist with this process as well in a
```
Comment (belisarius222) [inline arvo/ames/ames.md:357]: > in a similar fashion to peer discovery

I'd rephrase this sentence.  Packet relaying is used as part of peer discovery; it's not just similar to it.

File: arvo/ames/ames.md [hunk 2]
```diff
@@ -319,34 +334,45 @@ with `%boon`s and create congestion there. So naxplanations have their own bone,
...
+similar fashion to peer discovery.
+
+In the case of moons, their parent is responsible for packet relaying and peer
```
Comment (belisarius222) [inline arvo/ames/ames.md:360]: This is a nitpick, but I'd say `their parent planet`, in case the reader forgot the names for different kinds of nodes.

Comment (belisarius222) [reply arvo/ames/ames.md:360]: For cleaner grammar, I'd use singular 'moon' to match the singular 'parent':  `In the case of a moon, its parent is responsible` ...

File: arvo/ames/ames.md [hunk 3]
```diff
@@ -355,23 +381,28 @@ private key and the public key of `~worwel-sipnum`, obtained via Jael. Then
...
 intended recipient of the packet, and so gets ready to forward it to
-`~worwel-sipnum`. First, it [`+cue`s](@/docs/hoon/reference/stdlib/2p.md#cue)
-(deserializes) the payload and changes the `origin` to the IP and port of
-`~bacbel-tagfeb`. Then it `+jam`s `[origin content]` to form a new payload, and
-`+mug`s that payload to get a new checksum. It replaces the old payload and
-checksum with the new, and then sends the packet along to `~worwel-sipnum`.
+`~worwel-sipnum`. First, it replaces the origin field with the IP and port of
+`~bacbel-tagfeb`, then computes the `+mug` of the body and replaces the checksum
+with the new mug. Since the packet is being relayed, `~zod` flips the "is it
+relayed?" bit to 1. `~zod` then forwards the packet to `~worwel-sipnum`.
```
Comment (belisarius222) [inline arvo/ames/ames.md:399]: The 'relayed' bit is a loobean, so `relayed` is `&`, or `%.y`, which is 0, not 1.

File: arvo/ames/ames.md [hunk 4]
```diff
@@ -355,23 +381,28 @@ private key and the public key of `~worwel-sipnum`, obtained via Jael. Then
...
+
+In order to decrypt the packet, `~worwel-sipnum` will need to include the
```
Comment (belisarius222) [inline arvo/ames/ames.md:401]: The receiver doesn't mutate the packet to insert the AD vector.  It just needs to calculate the AD vector and pass that as an input to the decryption function along with the SIV and ciphertext.

**Review by belisarius222 — CHANGES_REQUESTED** [review]
Some minor edits.

**Comment by belisarius222** [comment]
Sorry for the delay in reviewing!

File: arvo/ames/ames.md (hunk 2 above)
Comment (drbeefsupreme) [reply arvo/ames/ames.md:360]: Since galaxies and stars can have moons, I didn't want to say planet. I changed it to this though:

```
In the case of a moon, its parent ship is responsible for packet relaying and peer
discovery, analagous to the role that galaxies currently play for stars and planets.
```

**Comment by matildepark** [comment]
Much apologies — you'll have to retarget this against https://github.com/urbit/urbit.org/tree/master/content/docs.
